### PR TITLE
firefox: use SHA512SUM in update.nix

### LINKED ISSUE #125662: Firefox sources update script error
**Describe the bug**
I was wanting to update the `firefox-devedition-bin-unwrapped` sources since they're quite out of date and so I tried running the update script, but it errored out with the log below. Perhaps the pubkey needs to be updated since it's now [hardcoded](https://github.com/NixOS/nixpkgs/commit/2b44f5640bf07a14dcb8c0dd5b8c410f205a4f09).

**To Reproduce**
1. `git clone https://github.com/nixos/nixpkgs.git`
2. `cd nixpkgs`
3. `nix-shell maintainers/scripts/update.nix --argstr package firefox-devedition-bin-unwrapped`

**Expected behavior**
Expected `pkgs/applications/networking/browsers/firefox-bin/devedition_sources.nix` to be updated.

**Screenshots**

**Additional context**
Error log:
```
--- SHOWING ERROR LOG FOR firefox-devedition-bin-unwrapped-85.0b6 ----------------------

+ pushd pkgs/applications/networking/browsers/firefox-bin
++ mktemp -d
+ HOME=/run/user/1000/tmp.cEVdQ0kylx
++ mktemp -d
+ export GNUPGHOME=/run/user/1000/tmp.vD7lNh2xBw
+ GNUPGHOME=/run/user/1000/tmp.vD7lNh2xBw
+ gpg --import /nix/store/rp5vgjkyz6naiga5fl7sfs9gzr6a78kh-mozilla.asc
gpg: keybox '/run/user/1000/tmp.vD7lNh2xBw/pubring.kbx' created
gpg: /run/user/1000/tmp.vD7lNh2xBw/trustdb.gpg: trustdb created
gpg: key 61B7B526D98F0353: public key "Mozilla Software Releases <release@mozilla.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
++ mktemp
+ tmpfile=/run/user/1000/tmp.1faj2ArsM0
+ url=http://archive.mozilla.org/pub/devedition/releases/
++ xidel -s http://archive.mozilla.org/pub/devedition/releases/ --extract //a
++ sed 's/.$//'
++ grep '^[0-9]'
++ grep -e 'b\([[:digit:]]\|[[:digit:]][[:digit:]]\)$'
++ grep -v funnelcake
++ sort --version-sort
++ tail -1
+ version=90.0b3
+ curl --silent -o /run/user/1000/tmp.cEVdQ0kylx/shasums http://archive.mozilla.org/pub/devedition/releases/90.0b3/SHA256SUMS
+ curl --silent -o /run/user/1000/tmp.cEVdQ0kylx/shasums.asc http://archive.mozilla.org/pub/devedition/releases/90.0b3/SHA256SUMS.asc
+ gpgv --keyring=/run/user/1000/tmp.vD7lNh2xBw/pubring.kbx /run/user/1000/tmp.cEVdQ0kylx/shasums.asc /run/user/1000/tmp.cEVdQ0kylx/shasums
gpgv: can't allocate lock for '/run/user/1000/tmp.vD7lNh2xBw/pubring.kbx'
gpgv: Signature made Thu 03 Jun 2021 10:48:30 PM UTC
gpgv:                using RSA key 4360FE2109C49763186F8E21EBE41E90F6F12F6D
gpgv: Can't check signature: No public key


--- SHOWING ERROR LOG FOR firefox-devedition-bin-unwrapped-85.0b6 ----------------------
```

**Notify maintainers**

@taku0 @lovesegfault

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.7, NixOS, 21.11pre292748.6933d068c5d (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210503_6d2553a`
 - channels(skykanin): `""`
 - channels(root): `"nixos-21.11pre292748.6933d068c5d"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 ```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by taku0**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/pull/121755#discussion_r626193224

`update.nix` of `firefox` use generic `update-source-version` to compute the hash.  This PR makes `update.nix` to use `SHA512SUMS` file provided by Mozilla.
This is based on `update.nix` of `firefox-bin`.

This also fixes `update.nix` for ESR version.

Fixes: #125662

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by taku0** [comment]
Updated PGP key. It expired 2021-05-29.
https://archive.mozilla.org/pub/firefox/releases/78.11.0esr/KEY

File: pkgs/applications/networking/browsers/firefox/update.nix [hunk 1]
```diff
@@ -15,7 +16,12 @@
 
 writeScript "update-${attrPath}" ''
   #!${runtimeShell}
-  PATH=${lib.makeBinPath [ common-updater-scripts coreutils curl gnugrep gnused xidel ]}
+  PATH=${lib.makeBinPath [ common-updater-scripts coreutils curl gnugrep gnupg gnused xidel ]}
+
+  set -eux
+  HOME=`mktemp -d`
+  export GNUPGHOME=`mktemp -d`
+  gpg --import ${../firefox-bin/mozilla.asc}
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/firefox/update.nix:24]: Can we just fetch the key from the web and add a hash here?

Comment (taku0) [reply pkgs/applications/networking/browsers/firefox/update.nix]: Changed to use `gpg --receive-keys`.

**Comment by Artturin** [comment]
Updating the key Closes #125662

**Comment by skykanin** [comment]
Tested this locally and firefox sources now update as expected after running the script. Fixes #125662. LGTM

**Comment by drupol** [comment]
Hi!

Thanks for your contribution but this PR is now obsolete since the changes has been made in https://github.com/NixOS/nixpkgs/pull/127067

Closing it.

**Pull request closed without merging**